pkg/apis/componentconfig/v1alpha1/types.go

```diff
@@ -21,39 +21,90 @@ import (
 	"k8s.io/kubernetes/pkg/api/v1"
 )
 
+// ClientConnectionConfiguration contains details for constructing a client.
+type ClientConnectionConfiguration struct {
+	// KubeConfigFile is the path to a kubeconfig file.
```

Comment:
I think the convention is to not follow the godoc and make these match the json tag so the field name shows up correctly in generate documentation.